### PR TITLE
fix(processor): Use flight do_exchange replace flight do_put

### LINKED ISSUE #6807: flake test: tpch test failure in cluster mode
There's a very strange failure of cluster test. Some of the TPCH test cases(e.g. q3, q7) can always pass on my Mac laptop, but will get stuck on Linux environment(both CI env and my own Ubuntu PC).

@zhang2014 Have you encountered the problem before? Seems all the runtime workers are blocked on the task queue and won't get scheduled for unknown reason.

_Originally posted by @leiysky in https://github.com/datafuselabs/databend/issues/6440#issuecomment-1194078380_

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

fix(processor): Use flight do_exchange replace flight do_put

Fixes #6807 #6730 #6591

### CODE REVIEW

File: query/tests/it/api/rpc/flight_client.rs [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+    Ok(())
+}
+
+// struct TestFlightService {}
+//
+// type FlightStream<T> = Pin<Box<dyn Stream<Item=std::result::Result<T, Status>> + Send + Sync + 'static>>;
+//
+// #[async_trait::async_trait]
+// impl FlightService for TestFlightService {
```
Comment (sundy-li) [inline query/tests/it/api/rpc/flight_client.rs:86]: git has histories,  better delete unused codes?

**Comment by Xuanwo** [comment]
Unit tests failed for timeout.

File: query/src/api/rpc/exchange/exchange_manager.rs [hunk 2]
```diff
@@ -80,24 +71,61 @@ impl DataExchangeManager {
...
-                exchange_senders.insert(target.clone(), exchange_sender);
+        let mut request_exchanges = vec![];
+        let mut targets_exchanges = HashMap::new();
+
+        let source = &packet.executor.id;
+        for connection_info in &packet.connections_info {
+            if connection_info.create_request_channel {
+                let query_id = &packet.query_id;
+                let address = &connection_info.target.flight_address;
+                let mut flight_client = Self::create_client(&self.config, address).await?;
```
Comment (BohuTANG) [inline query/src/api/rpc/exchange/exchange_manager.rs:82]: What's the client timeout number?

Comment (zhang2014) [reply query/src/api/rpc/exchange/exchange_manager.rs:82]: We use the default connection timeout and the fixed sending and receiving timeout (60 seconds). I have created a new issue to track this. #7092

**Review by BohuTANG — APPROVED** [review]
LGTM

**Review by sundy-li — APPROVED** [review]

**Pull request merged**